org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/GuardedPattern.java

```diff
@@ -213,6 +223,16 @@ public Expression getExpression() {
 	 */
 	public Pattern getPattern() {
 		supportedOnlyIn21();
+		if(this.pattern == null) {
+            // lazy init must be thread-safe for readers
+            synchronized (this) {
+                if(this.pattern == null) {
+                    preLazyInit();
+                    this.pattern = this.ast.newTypePattern();
```

Comment:
Please use:
                    this.pattern = this.ast.newNullPattern();

Reply:
`newTypePattern()` has been modified to `newNullPattern()`.